[Bug]: `InvalidPathException` on Windows caused by `BuildAgent`

### Brokk Version

0.13.4

### Operating System

Windows

### What happened?

This is a diagnosis of https://github.com/BrokkAi/brokk/issues/885.

It seems that the BuildAgent is attempting to create a path from a glob.

### Relevant log output

```shell
2025-09-06 14:29:58,668 [pool-12-thread-6] DEBUG ToolRegistry.executeTool:367 - Invoking tool 'reportBuildDetails' with args: [mvn --quiet compile, mvn --quiet test, mvn --quiet test -Dtest={{#classes}}{{value}}{{^-last}},{{/-last}}{{/classes}}, [generated-code/]]
2025-09-06 14:29:58,672 [pool-12-thread-6] ERROR ContextManager.lambda$ensureBuildDetailsAsync$56:1589 - Build Information Agent did not complete successfully (aborted or errored). Build details not saved. Error: java.lang.reflect.InvocationTargetException
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at io.github.jbellis.brokk.tools.ToolRegistry.executeTool(ToolRegistry.java:379) ~[brokk-0.13.4.jar:?]
	at io.github.jbellis.brokk.tools.ToolRegistry.executeTool(ToolRegistry.java:324) ~[brokk-0.13.4.jar:?]
	at io.github.jbellis.brokk.agents.BuildAgent.execute(BuildAgent.java:199) ~[brokk-0.13.4.jar:?]
	at io.github.jbellis.brokk.ContextManager.lambda$ensureBuildDetailsAsync$56(ContextManager.java:1580) ~[brokk-0.13.4.jar:?]
	at io.github.jbellis.brokk.ContextManager.lambda$submitBackgroundTask$53(ContextManager.java:1511) ~[brokk-0.13.4.jar:?]
	at io.github.jbellis.brokk.util.LoggingExecutorService.lambda$wrap$6(LoggingExecutorService.java:102) ~[brokk-0.13.4.jar:?]
	at io.github.jbellis.brokk.util.LoggingExecutorService.lambda$submit$0(LoggingExecutorService.java:34) ~[brokk-0.13.4.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.lang.reflect.InvocationTargetException
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at io.github.jbellis.brokk.tools.ToolRegistry.executeTool(ToolRegistry.java:368) ~[brokk-0.13.4.jar:?]
	... 11 more
Caused by: java.nio.file.InvalidPathException: Illegal char <*> at index 0: **/generated-sources/
	at sun.nio.fs.WindowsPathParser.normalize(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsPathParser.parse(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsPathParser.parse(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsPath.parse(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsFileSystem.getPath(Unknown Source) ~[?:?]
	at java.nio.file.Path.of(Unknown Source) ~[?:?]
	at io.github.jbellis.brokk.agents.BuildAgent.lambda$reportBuildDetails$2(BuildAgent.java:313) ~[brokk-0.13.4.jar:?]
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source) ~[?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
	at java.util.stream.ReferencePipeline.collect(Unknown Source) ~[?:?]
	at io.github.jbellis.brokk.agents.BuildAgent.reportBuildDetails(BuildAgent.java:315) ~[brokk-0.13.4.jar:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at io.github.jbellis.brokk.tools.ToolRegistry.executeTool(ToolRegistry.java:368) ~[brokk-0.13.4.jar:?]
	... 11 more
2025-09-06 14:29:58,673 [pool-12-thread-6] WARN  Chrome.toolError:804 - Build Information Agent did not complete successfully (aborted or errored). Build details not saved. Error: java.lang.reflect.InvocationTargetException: Build Information Agent failed
2025-09-06 14:29:58,673 [pool-12-thread-6] DEBUG Chrome.systemOutput:810 - Build details inferred and saved
```

--- a/app/src/main/java/io/github/jbellis/brokk/util/BuildToolConventions.java
+++ b/app/src/main/java/io/github/jbellis/brokk/util/BuildToolConventions.java
@@ -64,7 +64,7 @@ public class BuildToolConventions {
 
     public static List<String> getDefaultExcludes(BuildSystem system) {
         return switch (system) {
-            case MAVEN -> List.of("target/", "**/generated-sources/");
+            case MAVEN -> List.of("target/");
             case GRADLE -> List.of("build/", ".gradle/");
             case SBT -> List.of("target/"); // Scala target directory
             case NPM -> List.of("node_modules/", "dist/");
@@ -72,7 +72,7 @@ public class BuildToolConventions {
             case BAZEL -> List.of("bazel-out/");
             case CMAKE -> List.of("build/", "out/"); // Common for CMake and general Make
             case PYTHON ->
-                List.of("__pycache__/", ".pytest_cache/", ".mypy_cache/", ".tox/", "*.egg-info/", "build/", "dist/");
+                List.of("__pycache__/", ".pytest_cache/", ".mypy_cache/", ".tox/", ".egg-info/", "build/", "dist/");
             default -> List.of(); // UNKNOWN or any other unhandled system
         };
     }

